pkg/api/v1beta3/types.go

```diff
@@ -995,6 +1006,16 @@ type Binding struct {
 	Target ObjectReference `json:"target" description:"an object to bind to"`
 }
 
+// DeleteOptions may be provided when deleting an API object
+type DeleteOptions struct {
+	TypeMeta `json:",inline"`
```

Comment:
Kind and version are needed so that we can properly version this struct when it comes in from the API.  All objects have to be versioned.

I'll add ObjectMeta so this is consistent with Bindings.